src/runtime/Codecs/IterableDecoder.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+
+namespace Python.Runtime.Codecs
+{
+    public class IterableDecoder : IPyObjectDecoder
+    {
+        internal static bool IsIterable(Type targetType)
+        {
+            //if it is a plain IEnumerable, we can decode it using sequence protocol.
+            if (targetType == typeof(System.Collections.IEnumerable))
+                return true;
+
+            if (!targetType.IsGenericType)
+                return false;
+
+            return targetType.GetGenericTypeDefinition() == typeof(IEnumerable<>);
+        }
+
+        internal static bool IsIterable(PyObject objectType)
+        {
+            return objectType.HasAttr("__iter__");

```

Comment:
Can this be replaced with a call to `PyIter_Check`?

Reply:
Yes it can

Reply:
@lostmsu actually as it turns out, no it can't.  It leads to some test failures.
https://github.com/pythonnet/pythonnet/pull/1084/checks?check_run_id=1930584148

The reason is that the codec only has the *type*, not the *instance*.  PyIter_Check is meant to be called on the instance.

Reply:
Right. OK then.